src/modules/DASDController.rb

```diff
@@ -851,6 +791,106 @@ def report_error(headline, message, details)
         Yast2::Popup.show(message, headline: headline, details: details)
       end
     end
+
+    # Activates a disk if its not activated
+    #
+    # When the disk is already activated, it returns '8' if the
+    # disk is unformatted or '0' otherwise. The idea is to mimic
+    # the same API that ActivateDisk.
+    #
+    # @return [Boolean] Returns an error code (8 means 'unformatted').
+    def activate_disk_if_needed(channel, diag)
+      disk = find_disks.find { |d| d["channel"] == channel }
+      if disk && active_disk?(disk)
+        log.info "Disk #{channel} is already active. Skipping the activation."
+        return disk["formatted"] ? 0 : 8
+      end
+      ActivateDisk(channel, diag) unless disk && active_disk?(disk)
```

Comment:
structure is really strange. why it check again for active_disk? when it is done already above?

I found whole that `unless ...` part unnecessary and confusing.

Reply:
Oh, a leftover. Thanks for catching it.